.github/workflows/lint-nix.yml

```diff
@@ -0,0 +1,63 @@
+# This yml handles linting nix configs for each PR
+### Workflow
+# 1. Developer writes nix configs
+# 2. Developer pushes code up to fork and opens new PR
+# 3. This job kicks off, verifying the new configuration they added
+# 4. If the job suceeds, the PR is allowed to be merged. If not, the PR is unable to be merged
+
+name: Lint Nix Configs
+on: pull_request
+
+jobs:
+  linter:
+    name: Run nixpkgs-fmt
+    runs-on: ubuntu-18.04
+
+    # nixos/nix docker image v2.3 SHA
+    container:
+      image: nixos/nix@sha256:af330838e838cedea2355e7ca267280fc9dd68615888f4e20972ec51beb101d8
+
+    steps:
+      - name: Pulls Repository
+        uses: actions/checkout@v2
+      
+        # Utilizing an external GitHub Action to obtain all changed files
+        # https://github.com/jitterbit/get-changed-files
+        # Utilizing SHA for v1
+      - name: Get All Changed Files
```

Comment:
Since the `nix-fmt` is fast enough, lets just fmt the entire repo so we ensure consistency.

Reply:
Sure, I can see where you're coming from. I was going at it from the efficient perspective since you'd only care about any new files that come in (presuming the repo itself is formatted properly formatted). But as you said, `nixpkgs-fmt` is really fast so there isn't too much time wasted.

The only thing is that with the current logic, if I were to add that new check in, it will cause the check to fail since there's currently files existing in the repo that do not meet the formatting standards. In order to make this PR pass, this PR will have to re-add in those fixed files as well.